modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/test/java/com/liferay/dynamic/data/mapping/storage/constants/FieldConstantsTest.java

```diff
@@ -66,4 +71,20 @@ public void testGetSerializableReturnIntegerWithLocaleUS() {
 		Assert.assertEquals(Integer.valueOf(3), (Integer)serializable);
 	}
 
+	@Test
+	public void testGetSerializableReturnNumberArray() {
+		List<Serializable> values = Arrays.asList(
+			Integer.valueOf(1000), "10000000000");
+
+		Serializable serializable = FieldConstants.getSerializable(
+			FieldConstants.DOUBLE, values);
+
+		Assert.assertTrue(serializable instanceof Number[]);
+
+		Number[] numbers = (Number[])serializable;
+
+		Assert.assertEquals(Integer.valueOf(1000), numbers[0]);
+		Assert.assertEquals(new BigDecimal("10000000000"), numbers[1]);
```

Comment:
I was thinking of covering more cases to make sure INTEGER type has the same result as DOUBLE type and it doesn't matter whatever is the size of the number, if it's a String instance, it'll result in BigDecimal. Also, we don't need to use `valueOf` for doubles and integers. Let me know what you think. 🙂 
```suggestion
		List<Serializable> values = Arrays.asList(
			"1", 1.0, 1000, "10000000000");

		Serializable serializable = FieldConstants.getSerializable(
			FieldConstants.DOUBLE, values);

		Number[] numbers = (Number[])serializable;

		Assert.assertEquals(new BigDecimal("1"), numbers[0]);
		Assert.assertEquals(1.0, numbers[1]);
		Assert.assertEquals(1000, numbers[2]);
		Assert.assertEquals(new BigDecimal("10000000000"), numbers[3]);

		serializable = FieldConstants.getSerializable(
			FieldConstants.INTEGER, values);

		numbers = (Number[])serializable;

		Assert.assertEquals(new BigDecimal("1"), numbers[0]);
		Assert.assertEquals(1.0, numbers[1]);
		Assert.assertEquals(1000, numbers[2]);
		Assert.assertEquals(new BigDecimal("10000000000"), numbers[3]);
```

Reply:
Hi @rodrigopaulino, yes, it's a good idea to test more cases.